Additional jdk versions on ci build

npotluri-rh opened the pull request:
<!-- Please describe your pull request here. -->

### Testing done

<!-- Comment:
Provide a clear description of how this change was tested.
At minimum this should include proof that a computer has executed the changed lines.
Ideally this should include an automated test or an explanation as to why this change has no tests.
Note that automated test coverage is less than complete, so a successful PR build does not necessarily imply that a computer has executed the changed lines.
If automated test coverage does not exist for the lines you are changing, you must describe the scenario(s) in which you manually tested the change.
For frontend changes, include screenshots of the relevant page(s) before and after the change.
For refactoring and code cleanup changes, exercise the code before and after the change and verify the behavior remains the same.
-->
### Submitter checklist

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md in your own repository 
-->

npotluri-rh committed:
jdk11 and jdk17 added for ci

npotluri-rh committed:
fix compile error

npotluri-rh committed:
update versions

npotluri-rh committed:
few version fixes

npotluri-rh committed:
revert the pom changes

olivergondza commented on pom.xml:
```diff
@@ -102,13 +101,15 @@
                     </additionalOptions>
                 </configuration>
             </plugin>
+            <!-- Do not deploy anything except for the plugin -->
```
This comment refers to the reverted code. Please revert as well

olivergondza commented on pom.xml:
```diff
@@ -102,13 +101,15 @@
                     <compilerArgs>
                         <arg>-Xlint:all</arg>
                     </compilerArgs>
+                    <skip>true</skip>
```
This might be the same case. Not sure if compiler supports skipping, but we certainly do not want to do that.

pjanouse replied:
If I recall correctly, this was for the purpose that we wanted to get a report of all complier/linter issues (therefore 'Xlint:all') but there were also false issues which didn't have a proper solution/work-around. So for that purpose we couldn't stop the build because we have had only two options there:
1. Do all Lint reports but not fast-fail building process (because we couldn't fix all linted reports due to dependencies/transitive dependencies where things are out of our control)
2. Do a particular (probably none) Lint report and have fast-fail for that (if reported issues are really valid)

npotluri-rh committed:
revert build plugin changes

olivergondza committed:
Fix mock stubbing failures on Java 11

olivergondza committed:
Open modules that mockito needs for mock serialization

```
...
Caused by: java.lang.RuntimeException: Failed to serialize org.jenkinsci.plugins.proccleaner.PreBuildCleanup#cleaner for class org.jenkinsci.plugins.proccleaner.PreBuildCleanup
  at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:274)
  at hudson.util.RobustReflectionConverter$2.visit(RobustReflectionConverter.java:241)
  at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:174)
  at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:226)
  at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:163)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:87)
  at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeBareItem(AbstractCollectionConverter.java:94)
  at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:66)
  at hudson.util.DescribableList$ConverterImpl.marshal(DescribableList.java:274)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
  at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:283)
  at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:270)
  ... 36 more
Caused by: java.lang.RuntimeException: Failed to serialize org.mockito.internal.creation.bytebuddy.ByteBuddyCrossClassLoaderSerializationSupport#mutex for class org.mockito.internal.creation.bytebuddy.ByteBuddyCrossClassLoaderSerializationSupport
  at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:274)
  at hudson.util.RobustReflectionConverter$2.visit(RobustReflectionConverter.java:241)
  at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:174)
  at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:226)
  at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:163)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:87)
  at com.thoughtworks.xstream.converters.reflection.SerializableConverter$1.defaultWriteObject(SerializableConverter.java:212)
  at com.thoughtworks.xstream.converters.reflection.SerializableConverter.doMarshal(SerializableConverter.java:272)
  at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:90)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:44)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:87)
  at com.thoughtworks.xstream.converters.reflection.SerializableConverter$1.defaultWriteObject(SerializableConverter.java:212)
  at com.thoughtworks.xstream.converters.reflection.SerializableConverter.doMarshal(SerializableConverter.java:272)
  at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:90)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
  at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:283)
  at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:270)
  ... 52 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private static final long java.util.concurrent.locks.ReentrantLock.serialVersionUID accessible: module java.base does not "opens java.util.concurrent.locks" to unnamed module @4116aac9
  at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
  at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
  at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
  at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
  at com.thoughtworks.xstream.converters.reflection.FieldDictionary.buildDictionaryEntryForClass(FieldDictionary.java:176)
  at com.thoughtworks.xstream.converters.reflection.FieldDictionary.buildMap(FieldDictionary.java:142)
  at com.thoughtworks.xstream.converters.reflection.FieldDictionary.fieldsFor(FieldDictionary.java:80)
  at com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider.visitSerializableFields(PureJavaReflectionProvider.java:167)
  at hudson.util.RobustReflectionConverter.doMarshal(RobustReflectionConverter.java:206)
  at hudson.util.RobustReflectionConverter.marshal(RobustReflectionConverter.java:163)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:68)
  at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:59)
  at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:83)
  at hudson.util.RobustReflectionConverter.marshallField(RobustReflectionConverter.java:283)
  at hudson.util.RobustReflectionConverter$2.writeField(RobustReflectionConverter.java:270)
```
